### PR TITLE
deploy_template() shortcut allows for autoskip

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: cfme/tests/control/test_actions.py -v --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 00e393d9b55fe3bf8f24d8f17c42b7a448e43fd1:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
This seems to not break anything, if this is something we'd use more, then merge :)

File: utils/virtual_machines.py [hunk 1]
```diff
@@ -44,30 +47,35 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
...
+                        logger.info('VM/instance %s destroyed', vm_name)
+                    else:
+                        logger.error('Error destroying VM/instance %s', vm_name)
+            except Exception as f:
+                logger.error(
+                    'Could not destroy VM/instance {} ({}: {})'.format(
+                        vm_name, type(f).__name__, str(f)))
+            finally:
+                raise e
+    except RHEVRequestError as e:
```
Comment (seandst) [inline utils/virtual_machines.py]: Maybe change the name, but I think I'd rather see `allow_skip` to be a list of exceptions that trigger a skip, or maybe a list of callables that return True/False if a skip should happen based on the exception, or maybe both, or something else you devise. :)

**Comment by seandst** [comment]
Also, any ideas for not causing the `RHEVRequestError` in the first place?

**Comment by mfalesni** [comment]
@seandst rhev likes to complain about not having enough space to provision and so (error 409 and similar). I am really desperate how to solve this to not fail.

**Comment by mfalesni** [comment]
@seandst I made it so you can pass an iterable, a dict with keys as exception classes and callables and also you can pass "default" which will pick the most common errors we are having at the moment (RHEV's RequestError so far).

**Comment by dajoRH** [comment]
Lint report for commit c24f017ba316323a7d809ba4f4bacc4142f816ef:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by seandst** [comment]
This is pretty great. Assuming PRT is happy with it, then :+1:. Be on the lookout for other place we can apply this to help cut down on some of the deployment error noise.

**Comment by seandst** [comment]
Aside from rhevm34 being sad, everything worked.

**Pull request merged**